src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -189,7 +191,59 @@ public static Invocation CreateInvocation(InvocationVersionOne v1Invocation, Run
             return invocation;
         }
 
-        public static LogicalLocation CreateLogicalLocation(LogicalLocationVersionOne v1LogicalLocation)
+        internal Location CreateLocation(LocationVersionOne v1Location)
+        {
+            Location location = null;
+
+            if (v1Location != null)
+            {
+                location = new Location
+                {
+                    FullyQualifiedLogicalName = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName,
+                    PhysicalLocation = CreatePhysicalLocation(v1Location.ResultFile),
+                    Properties = v1Location.Properties
+                };
+
+                location.SetProperty($"{FromPropertyBagPrefix}/analysisTarget", v1Location.AnalysisTarget);
+                location.SetProperty($"{FromPropertyBagPrefix}/decoratedName", v1Location.DecoratedName);
+            }
+
+            return location;
+        }
+
+        internal Location CreateLocation(string fullyQualifiedLogicalName,
+                                         string logicalLocationKey,
+                                         string message,
+                                         Uri uri,
+                                         string uriBaseId,
+                                         int column,
+                                         int line)
+        {
+            var location = new Location
+            {
+                Message = CreateMessage(message)
+            };
+
+            if (string.IsNullOrWhiteSpace(logicalLocationKey) &&
+                !string.IsNullOrWhiteSpace(fullyQualifiedLogicalName))
+            {
+                LogicalLocation logicalLocation = CreateLogicalLocation(fullyQualifiedLogicalName);
+                location.FullyQualifiedLogicalName = AddLogicalLocation(logicalLocation);
+            }
```

Comment:
Bug: If they are both non-null, then `location.FQLN` should get `logicalLocationKey`.

Reply:
Fixed

Reply:
Yeah I've gone round and round on that subject